buddy_alloc.h

```diff
@@ -917,16 +1015,19 @@ static struct buddy_tree_pos deepest_position_for_offset(struct buddy *buddy, si
 }
 
 static struct buddy_tree_pos position_for_address(struct buddy *buddy, const unsigned char *addr) {
+    struct buddy_tree *tree;
+    struct buddy_tree_pos pos;
+
     /* Find the deepest position tracking this address */
     unsigned char *main = buddy_main(buddy);
     ptrdiff_t offset = addr - main;
 
-    if (offset % buddy->alignment) {
+    if (offset % (ptrdiff_t)buddy->alignment) {
```

Comment:
offset is ptrdiff_t(signed), alignement is unsigned. I think the alignment should be within PTRDIFF_MAX.

Reply:
Offset is signed but it can can never be negative. And casting the `buddy->alignment` to `(ptrdiff_t)` is semantically wrong as the buddy->alignment is not a ptrdiff_t by any stretch of reasoning - ptrdiff_t should only be used for pointer differences